modules/custom/dkan_datastore/src/Storage/JobStore.php

```diff
@@ -47,6 +47,45 @@ public function get(string $uuid, string $jobClass) {
     }
   }
 
+  /**
+   * Retrieve all.
+   */
+  public function retrieveAll(string $jobClass): array {
+    $tableName = $this->getTableName($jobClass);
+
+    $this->validateJobClassAndTableExistence($jobClass, $tableName);
+
+    $result = $this->connection->select($tableName, 't')
+      ->fields('t', ['ref_uuid', 'job_data'])
+      ->execute()
+      ->fetchAll();
+
+    if ($result === FALSE) {
+      throw new \Exception("No data in table: $tableName");
+    }
+
+    return array_reduce($result, function ($carry, $item) use ($jobClass) {
+      $job = $jobClass::hydrate($item->job_data);
+      if (isset($job) && ($job instanceof $jobClass)) {
+        $carry[$item->ref_uuid] = $job;
+      }
+      return $carry;
+    }, []);
+  }
```

Comment:
It doesn't seem to trigger Drupal coding standards, and I'm not entirely sure why, but I have a very minor nit-pick: It seems off that the anonymous function's ending curly braces aligns with `return`, so consider revisiting the position / indentation of the three arguments to maybe:

```
    return array_reduce(
      $result,
      function ($carry, $item) use ($jobClass) {
        $job = $jobClass::hydrate($item->job_data);
        if (isset($job) && ($job instanceof $jobClass)) {
          $carry[$item->ref_uuid] = $job;
        }
        return $carry;
      },
      []
    );
```